contracts/modules/cover/Cover.sol

```diff
@@ -665,15 +681,21 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard, Mu
     // increase amountPaidOut only *after* you read the segment
     cover.amountPaidOut += payoutAmountInAsset.toUint96();
 
-    IStakingProducts _stakingProducts = stakingProducts();
-    for (uint i = 0; i < allocations.length; i++) {
-      PoolAllocation memory allocation = allocations[i];
+    for (uint i = 0; i < burnedSegmentAllocations.length; i++) {
+      PoolAllocation memory allocation = burnedSegmentAllocations[i];
+
+      // Deallocate to free up capacity from the currently active / latest segment
+      // proportionately with payoutAmountInAsset / lastSegment.amount
+      uint deallocationAmountInNXM =
+        lastSegmentAllocations[i].coverAmountInNXM * payoutAmountInAsset / lastSegment.amount;
 
-      uint deallocationAmountInNXM = allocation.coverAmountInNXM * payoutAmountInAsset / segment.amount;
-      uint burnAmountInNxm = deallocationAmountInNXM * GLOBAL_CAPACITY_DENOMINATOR / segment.globalCapacityRatio;
+      // Burn the stake proportionally from the staking pool allocated for the target segment
+      // proportionately with payoutAmountInAsset / segment.amount
+      uint burnAmountInNxm =
+        payoutAmountInAsset * allocation.coverAmountInNXM * GLOBAL_CAPACITY_DENOMINATOR / segment.amount / segment.globalCapacityRatio;
```

Comment:
the issue here is some pools may be removed and new ones added so cannot assume it's the same one at the same index as the previous segment

Need to be processed into 2 diferent loops